lib/page/home/home_view_model.dart

```diff
@@ -3,29 +3,44 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:intl/intl.dart';
 import 'package:rxdart/subjects.dart';
 import 'package:survey/model/survey_model.dart';
+import 'package:survey/model/user_model.dart';
 import 'package:survey/page/home/home_state.dart';
 import 'package:survey/usecase/base/base_use_case.dart';
 import 'package:survey/usecase/get_cached_surveys_use_case.dart';
 import 'package:survey/usecase/get_surveys_use_case.dart';
+import 'package:survey/usecase/get_user_use_case.dart';
 
 const _homeCurrentDatePattern = 'EEEE, MMMM dd';
 const _surveysPageSize = 5;
 
 class HomeViewModel extends StateNotifier<HomeState> {
+  final GetUserUseCase _getUserUseCase;
   final GetSurveysUseCase _getSurveysUseCase;
   final GetCachedSurveysUseCase _getCachedSurveysUseCase;
 
   HomeViewModel(
+    this._getUserUseCase,
     this._getSurveysUseCase,
     this._getCachedSurveysUseCase,
   ) : super(const HomeState.init());
 
   int _surveysPageNumber = 1;
 
+  final BehaviorSubject<UserModel> _user = BehaviorSubject();
+
+  Stream<UserModel> get user => _user.stream;
+
   final BehaviorSubject<List<SurveyModel>> _surveys = BehaviorSubject();
 
   Stream<List<SurveyModel>> get surveys => _surveys.stream;
 
+  void getUser() async {
+    final result = await _getUserUseCase.call();
+    if (result is Success<UserModel>) {
```

Comment:
Are we missing handling the `Failed` case? 🤔

Reply:
@markgravity Since we don't emit anything if the `getUserUseCase` returns a failed result I'm not sure if we need the failed test case 🤔